WooCommerce/Classes/Authentication/Store Creation/Profiler/StoreCreationProfilerQuestionContainerView.swift

```diff
@@ -0,0 +1,95 @@
+import SwiftUI
+
+/// Hosting controller for `StoreCreationProfilerQuestionContainerView`
+final class StoreCreationProfilerQuestionContainerHostingController: UIHostingController<StoreCreationProfilerQuestionContainerView> {
+    init(viewModel: StoreCreationProfilerQuestionContainerViewModel, onSupport: @escaping () -> Void) {
+        super.init(rootView: StoreCreationProfilerQuestionContainerView(viewModel: viewModel, onSupport: onSupport))
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureTransparentNavigationBar()
+    }
+}
+
+/// Container view for the profiler questions of the store creation flow.
+struct StoreCreationProfilerQuestionContainerView: View {
+
+    @ObservedObject private var viewModel: StoreCreationProfilerQuestionContainerViewModel
+    private var onSupport: () -> Void
+
+    init(viewModel: StoreCreationProfilerQuestionContainerViewModel, onSupport: @escaping () -> Void) {
+        self.viewModel = viewModel
+        self.onSupport = onSupport
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            ProgressView(value: viewModel.currentQuestion.progress)
+                .progressViewStyle(.linear)
+
+            switch viewModel.currentQuestion {
+            case .sellingStatus:
+                StoreCreationSellingStatusQuestionContainerView(storeName: viewModel.storeName) { answer in
+                    viewModel.saveSellingStatus(answer)
+                } onSkip: {
+                    viewModel.saveSellingStatus(nil)
+                }
+            case .category:
+                StoreCreationCategoryQuestionView(viewModel: .init(storeName: viewModel.storeName, onContinue: { answer in
+                    viewModel.saveCategory(answer)
+                }, onSkip: {
+                    viewModel.saveCategory(nil)
+                }))
+            case .country:
+                StoreCreationCountryQuestionView(viewModel: .init(storeName: viewModel.storeName, onContinue: { answer in
+                    viewModel.saveCountry(answer)
+                }, onSupport: onSupport))
+            case .challenges:
+                StoreCreationChallengesQuestionView(viewModel: .init(onContinue: { answer in
+                    viewModel.saveChallenges(answer)
+                }, onSkip: {
+                    viewModel.saveChallenges([])
+                }))
+            case .features:
+                StoreCreationFeaturesQuestionView(viewModel: .init(onContinue: { answer in
+                    viewModel.saveFeatures(answer)
+                }, onSkip: {
+                    viewModel.saveFeatures([])
+                }))
+            }
+        }
```

Comment:
❓ non-blocking: Is it possible to animate this view change like a navigation controller's push view controller animation?

Reply:
Good thinking! I added `withAnimation` when we change the questions in be55f422a08be00fec3dced6b40aa9ae03ad59d0 - the view should has animations now:


https://github.com/woocommerce/woocommerce-ios/assets/5533851/d84b03b9-9746-4fa3-a629-a0bd9a313e6a

 The code looks quite repeated though - let me know if you know of better solution to this.